modules/tfm/tfm/boards/common/boot_hal.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2019-2020, Arm Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+#include <stdio.h>
+#include "cmsis.h"
+#include "region.h"
+#include "target_cfg.h"
+#include "boot_hal.h"
+#include "Driver_Flash.h"
+#include "flash_layout.h"
+#include "nrf_cc3xx_platform.h"
+#include "nrf_cc3xx_platform_kmu.h"
+
+/* Flash device name must be specified by target */
+extern ARM_DRIVER_FLASH FLASH_DEV_NAME;
+
+#if 0
+REGION_DECLARE(Image$$, ER_DATA, $$Base)[];
+REGION_DECLARE(Image$$, ARM_LIB_HEAP, $$ZI$$Limit)[];
+
+__attribute__((naked)) void boot_clear_bl2_ram_area(void)
+{
+    __ASM volatile(
+#ifndef __ICCARM__
+        ".syntax unified                             \n"
+#endif
+        "movs    r0, #0                              \n"
+        "subs    %1, %1, %0                          \n"
+        "Loop:                                       \n"
+        "subs    %1, #4                              \n"
+        "blt     Clear_done                          \n"
+        "str     r0, [%0, %1]                        \n"
+        "b       Loop                                \n"
+        "Clear_done:                                 \n"
+        "bx      lr                                  \n"
+        :
+        : "r" (REGION_NAME(Image$$, ER_DATA, $$Base)),
+          "r" (REGION_NAME(Image$$, ARM_LIB_HEAP, $$ZI$$Limit))
+        : "r0", "memory"
+    );
+}
+#endif
+
+int32_t boot_platform_init(void)
+{
+    int32_t result = 0;
+
+    /* Initialize the flash driver */
+    result = FLASH_DEV_NAME.Initialize(NULL);
+    if (result != ARM_DRIVER_OK) {
+        return 1;
+    }
+
+    /* Initialize the nrf_cc3xx runtime */
+    result = nrf_cc3xx_platform_init();
+    if (result != NRF_CC3XX_PLATFORM_SUCCESS) {
+        return 1;
+    }
+
+    return result;
+}
+
+void boot_platform_quit(struct boot_arm_vector_table *vt)
+{
+    /* Clang at O0, stores variables on the stack with SP relative addressing.
+     * When manually set the SP then the place of reset vector is lost.
+     * Static variables are stored in 'data' or 'bss' section, change of SP has
+     * no effect on them.
+     */
+    static struct boot_arm_vector_table *vt_cpy;
+    int32_t result;
+
+    /* Deinitialize the nrf_cc3xx runtime */
+    result = nrf_cc3xx_platform_deinit();
+    if (result != NRF_CC3XX_PLATFORM_SUCCESS)
+    {
```

Comment:
Different brace style here.

Reply:
This NITS will be solved after TF-M integration branch goes in